### PR TITLE
Order tasks page searching

### PR DESCRIPTION
**Pull request by shaobinspire**

### CODE REVIEW

**Review by kranar — COMMENTED** [review]
Needs to be tested.

**Comment by shaobinspire** [comment]
@kranar I have added some unit test cases for OrderTaskMatch, MatchCache and TableMatchCache.

File: Applications/Spire/Source/SpireTester/OrderTaskMatchTester.cpp [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+      REQUIRE(!matches(destination, "AS"));
+      REQUIRE(!matches(destination, "amexa"));
+      REQUIRE(!matches(destination, "NYS "));
+      REQUIRE(!matches(destination, "NYSEMKT"));
+      REQUIRE(!matches(destination, "LLC"));
+      REQUIRE(!matches(Destination(), "AMEX"));
+    }
+  }
+
+  TEST_CASE("match_order_typ") {
```
Comment (kranar) [inline Applications/Spire/Source/SpireTester/OrderTaskMatchTester.cpp:187]: typo

File: Applications/Spire/Source/SpireTester/TableMatchCacheTester.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+TEST_SUITE("TableMatchCache") {
+  TEST_CASE("match_cache") {
+    auto call_first_matcher = false;
+    auto call_second_matcher = false;
+    auto reset_call_matcher = [&] {
+      call_first_matcher = false;
+      call_second_matcher = false;
+    };
+    auto is_hitted_first_cache = [&] {
```
Comment (kranar) [inline Applications/Spire/Source/SpireTester/TableMatchCacheTester.cpp:25]: Rename ```hitted``` to ```hit```.

**Review by kranar — CHANGES_REQUESTED** [review]
Looks great.

**Pull request merged**